src/query/functions/utils/heap.go

```diff
@@ -68,21 +68,25 @@ func NewFloatHeap(isMaxHeap bool, capacity int) FloatHeap {
 		capacity = 0
 	}
 
+	if capacity > 10_000 {
```

Comment:
Is it used only by take/topk? Some comment describing the limit would be useful.

Reply:
Yes, it is used only by topk, bottomk. Ideally, this limit value could be taken somehow from `QueryContextOptions` but for this to happen much larger refactoring needs to be done and I didn't think it was worth it for now.

Reply:
This is a good question. Why did we need to add this limit, and how did we live without it before?
And why 10k? This is `O(nlogn)` data structure, it can deal with way more elements easily.

Reply:
10k is our current default limit for max time series to return. 
Previously, this would OOM on the following test: `eval_ordered instant at 50m bottomk(9999999999, http_requests{job="app-server",group="canary"})`

Reply:
Oh, so you should fix this by not allocating `k` size heap, but `min(k, len(input))` instead.